pkg/model/components/kubecontrollermanager_test.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"testing"
+	"time"
+
+	api "k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+)
+
+type ClusterParams struct {
+	CloudProvider     string
+	KubernetesVersion string
+	UpdatePolicy      string
+}
+
+func buildCluster(clusterArgs interface{}) *api.Cluster {
```

Comment:
I think you can just build the cluster, and then override the values.  You don't have any dependencies AFAICT.